### ISSUE TITLE
Improper zooming behavior

### ISSUE THREAD

**Issue by manu-st** [highcharts/highcharts#7994]
#### Expected behaviour
Zooming should not change the y-axis mapping for all y-axis.

#### Actual behavior
The y-axis are not matching

#### Live demo with steps to reproduce
Start with:
http://jsfiddle.net/o516oo0s/11/

You should get:
![image](https://user-images.githubusercontent.com/13194660/37308577-b9ed6862-25fb-11e8-9230-c9d6b2758f25.png)

Then drag a zoom selection area around points of the 2 series at position 120 and 150:
![image](https://user-images.githubusercontent.com/13194660/37308622-d4b8deb0-25fb-11e8-911a-a800d0c5d52a.png)

After zooming, they should still on the same horizontal line but depending on how the zoom area is performed, they are not always aligned:
![image](https://user-images.githubusercontent.com/13194660/37308679-058eb44c-25fc-11e8-9c2f-b65ae32e5cfc.png)

#### Product version
Using latest version

#### Affected browser(s)
All

**Comment by pawelfus** [comment]
Hi @manu-st 

This is the result of [`yAxis.startOnTick`](https://api.highcharts.com/highcharts/yAxis.startOnTick), [`yAxis.endOnTick`](https://api.highcharts.com/highcharts/yAxis.endOnTick) and [`chart.alignTicks`](https://api.highcharts.com/highcharts/chart.alignTicks) options and is expected behaviour. When you zoom in, your selection y-range can be for example 143-167. By default we have these options **enabled**, which may cause rendering bigger range than selected (e.g. 143->140 and 167->180). The same happens with the second yAxis and then we match both ticks. Matching ticks can increase the given range even more, in result changing extremes and ticks. You can disable these options and get this result: http://jsfiddle.net/BlackLabel/o516oo0s/24/ (note: [`getExtremsFromAll`](https://api.highcharts.com/highcharts/series.line.getExtremesFromAll) is enabled true).

Alternatively, you can use [`chart.events.selection`](https://api.highcharts.com/highcharts/chart.events.selection) event and modify extremes before these are applied on a chart.

**Comment by manu-st** [comment]
@pawelfus Thanks for the suggestion. I've modified my code to use `chart.events.selection` to the the extremes and it seems that everything is done properly as long as the `yAxis.startOnTick`, `xAxis.endOnTick` and `alignTicks` are set to false. If it is not set, then it is misbehaving.

However there is one small issue, when the graph starts with:
![image](https://user-images.githubusercontent.com/13194660/37385687-453c5010-2712-11e8-9ded-ad3c14fcf19f.png)

Then after zooming I get:
![image](https://user-images.githubusercontent.com/13194660/37385701-653a52f4-2712-11e8-9f4b-d3d7baf801c3.png)

Note that the lines are stopping at the last point in the visible part of the series of the zoomed area. I was expecting them to continue to connect to the next point not visible anymore. To be more explicit, the red and black series should have a line that is drawn up to the bottom axis and not stopping before.

**Comment by manu-st** [comment]
@pawelfus Could you clarify what is `getExtremesFromAll` as the documentation is quite confusing and how it is important here ? I tried with and without and did not see any visible effect.

**Comment by pawelfus** [comment]
@manu-st Could you recreate issue with missing lines? That shouldn't happen. Maybe you have a lot of points and [`series.cropThreshold`](https://api.highcharts.com/highcharts/series.line.cropThreshold) is removing some of them?

Regarding `getExtremesFromAll`: we calculate extremes **for a series** using only points within a range, unless `getExtremesFromAll` is set to true. If enabled, then we calculate extremes from all y-values. In your case, it depends on a dataset: if points outside zoomed range are close by, then it won't be a problem, but if values fluctuate a lot (e.g. 1-100-5-1000) then it's better (in my opinion) to get extremes from all points. TickPositions will be calculated for all points and rendered only for a visible range.

**Comment by manu-st** [comment]
I've put a dataset that exhibit the behavior: http://jsfiddle.net/fhdLxa21/ Zoom on the black / red lines and you should see the problem.

**Comment by pawelfus** [comment]
Thank you for the sample, please try `series.cropThreshold` option, it should resolve the issue (unfortunately I can not edit your demo in jsFiddle, it takes forever to type a single letter :( )

**Comment by manu-st** [comment]
I tried it but it did not change the behavior? I tried various values and nothing work. In the example, I sent the series have at most 2533 data point, so I set the cropThreshold to 2534.

Since we are already using Boost, I decided to try without boost and surprisingly it works properly with boost. So it seems that boost is causing our zoom issue. And it works regardless of the value for `cropThreshold`.
